docs/modules/mssql.md

```diff
@@ -477,4 +479,61 @@ curl -X PUT \
     }
 }
 '
-```
\ No newline at end of file
+```
+
+### Service: azure-sql-12-0-dbms-registered
+
+It is to *register* an existing Azure SQL Server as a SQL DBMS service instance. Both **azure-sql-12-0-database** service and **azure-sql-12-0-database-from-existing** service can be its child service.
+
+#### Behaviors
+
+##### Provision
+
+Please refer to [**azure-sql-12-0-dbms** service](#service-azure-sql-12-0-dbms) with extra required provisioning parameters below.
+
+###### Provisioning Parameters
+
+| Parameter Name | Type | Description | Required | Default Value |
+|----------------|------|-------------|----------|---------------|
+| `server` | `string` | The SQL server name. | Y | |
+| `administratorLogin` | `string` | The administratorLogin input when creating the SQL server. | Y | |
+| `administratorLoginPassword` | `string` | The administratorLoginPassword input when creating the SQL server. | Y | |
+
+##### Update
+
+Update the `administratorLogin` and/or `administratorLoginPassword` as they may change and the server is assumed to be managed by yourself.
+
+###### Updating Parameters
+
+| Parameter Name | Type | Description | Required | Default Value |
+|----------------|------|-------------|----------|---------------|
+| `administratorLogin` | `string` | New administratorLogin. | N | |
+| `administratorLoginPassword` | `string` | New administratorLoginPassword. | N | |
+
+##### Bind, Unbind
+
+Please refer to [**azure-sql-12-0-dbms** service](#service-azure-sql-12-0-dbms).
+
+##### Deprovision
+
+Do nothing. The SQL server would not be deleted in Azure.
+
+### Service: azure-sql-12-0-database-from-existing
+
+It is to create SQL database service instance from existing Azure SQL Database *for taking over the database*. Both **azure-sql-12-0-dbms** service and **azure-sql-12-0-dbms-registered** service can be its parent service.
```

Comment:
Do we need to have both parent services? 

The `azure-sql-12-0-dbms` is managed by OSBA. If you're going to register some pre-existing database, is that really a scenario that you'll be encountering?

Reply:
Oh, I mean "either" here. I'll polish the words...

Reply:
The question is: will there have a `azure-sql-12-0-database-from-existing` instance whose parent is `azure-sql-12-0-dbms`? If so, it indicates the user provisions a `azure-sql-12-0-dbms` instance using OSBA and then creates a database on it outside OSBA. Will we meet this scenario?

Reply:
@norshtein this is very rare but valid.

Reply:
Yeah, that is my question. I feel like the more common scenario is existing DBMS and we want to provision new databases on that existing DBMS.  

An existing database probably exists on an existing DBMS. In that case, do the CF users really want to manage the DBMS and the DB via OSBA? Or do they want to _use_ the database via OSBA through normal binding operations?

We should maybe draw out exactly what scenarios are going to be supported? There are lots of different combinations, I'm not sure we should support all of them due to complexity.

Maybe these?

* OSBA provisions DBMS and DB (separately or together)
* User provisions DBMS, registers it with OSBA
* User uses OSBA to provision new databases on previously registered DBMS
* User registers existing database (can that simply be a stand alone service?)

Reply:
> @norshtein this is very rare but valid.

Have you actually had users _ask_ for this use case? Or is this a case where you are just trying to cover the possibilities? I think it's a pretty odd use case tbqh.

Reply:
I prefer just following the statement of `azure-sql-12-0-database`. The detail is confusing.

Reply:
If I understand this correctly, after user registers an dbms to OSBA, it's **nearly the same** between `azure-sql-12-0-dbms` and `azure-sql-12-0-dbms-registered`, except the behavior in deprovisioning is different. So we may not need to clarify every use case in the document.

So from user's perspective, there is no difference between creating a database on  `azure-sql-12-0-dbms` and creating a database on `azure-sql-12-0-dbms-registered`, the only thing they need to do is to specify `parentAlias`, regardless of the real type of parent instance. 

And for `azure-sql-12-0-database-from-existing`, similar logic comes: though it's very odd to make `azure-sql-12-0-dbms` to be the parent of `azure-sql-12-0-database-from-existing`, but it's valid, just as you can make `azure-sql-12-0-dbms-registered` to be the parent of `azure-sql-12-0-database-from-existing`.

Reply:
@norshtein that's correct! Now we consider how to describe it brief... as someone didn't understand PostgreSQL doc, the simpler one correctly and opened an issue.